datafusion/optimizer/src/limit_push_down.rs

```diff
@@ -192,6 +196,18 @@ impl OptimizerRule for LimitPushDown {
             LogicalPlan::Join(join) => {
                 let limit = fetch + skip;
                 let new_join = match join.join_type {
+                    JoinType::Left | JoinType::Right | JoinType::Full
+                        if is_no_join_condition(join) =>
+                    {
+                        // push left and right
+                        push_down_join(join, Some(limit), Some(limit))
+                    }
+                    JoinType::LeftSemi | JoinType::LeftAnti
+                        if is_no_join_condition(join) =>
+                    {
+                        // push left
+                        push_down_join(join, Some(limit), None)
+                    }
                     JoinType::Left => push_down_join(join, Some(limit), None),
                     JoinType::Right => push_down_join(join, None, Some(limit)),
```

Comment:
remove these

Reply:
I think those are the lesser restrictive versions of push down (only to left/right side)

Reply:
> remove these

```
JoinType::Left | JoinType::Right | JoinType::Full
                        if is_no_join_condition(join) =>
```

the `on condition` is empty, the rule can be applied, but if the `on condition` is not empty, 

```
     JoinType::Left => push_down_join(join, Some(limit), None),
                    JoinType::Right => push_down_join(join, None, Some(limit)),
```
can be applied.

cc @jackwener

Reply:
Understanded it